src/infra/flow-system-inmem/src/services/flow/flow_service_inmem.rs

```diff
@@ -717,6 +723,47 @@ impl FlowServiceInMemory {
 
         Ok(())
     }
+
+    fn get_logical_plan_options(&self, flow_key: &FlowKey) -> LogicalPlanOptions {
+        let mut logical_plan_opts = LogicalPlanOptions::default();
+        if let FlowKey::Dataset(dataset_flow) = flow_key {
```

Comment:
Where do you check the flow type is compacting and not update ?

Reply:
Yeah, missed it, added.